frontend/src/user/profiles/UserProfile.tsx

```diff
@@ -0,0 +1,197 @@
+  following?: FollowingRecord[];
+}
+
+export default function UserProfile() {
+  const { user_id } = useParams();
+  const [cookies, _] = useCookies();
+  const dispatch = useDispatch();
+  const state = useSelector((state: RootState) => state);
+  const [user, setUser] = useState<SpecificUser>();
+  const [isViewingFollowers, setIsViewingFollowers] = useState(true);
+  const isFollowing = JSON.parse(
+    localStorage.getItem('user_following') || '{}'
+  ).find((following: FollowingRecord) => {
+    return following.following[0].id === parseInt(user_id ?? '');
+  });
+
+  const _getUser = (user_id: number) => {
+    getUser(user_id)
+      .then((response) => {
+        const { id, name, email, user_image } = response.data.data;
+        setUser({
+          id: id,
+          name: name,
+          email: email,
+          user_image: user_image,
+        });
+      })
+      .catch(() => dispatch(setIsError(true)));
+  };
+
+  const _getFollowersOfUser = (user_id: number) => {
+    getUserFollowers(user_id)
+      .then((response) => {
+        const user_followers = response.data.data.filter(
+          (follower: FollowerRecord) => {
+            return follower.follower[0].id !== cookies.user['id'];
+          }
+        );
+        setUser((value) => ({ ...value, followers: user_followers }));
+        dispatch(setIsLoading(false));
+      })
+      .catch();
+  };
+
+  const _getFollowingOfUser = (user_id: number) => {
+    getUserFollowing(user_id)
+      .then((response) => {
+        const user_following = response.data.data.filter(
+          (following: FollowingRecord) => {
+            return following.following[0].id !== cookies.user['id'];
+          }
+        );
+        setUser({ ...user, following: user_following });
+      })
+      .catch();
+  };
+
+  const handleFollowUser = (user_id: number) => {
+    dispatch(setIsLoading(true));
+    const request = isFollowing
+      ? unfollowUser(cookies.user['id'], user_id)
+      : followUser(cookies.user['id'], { following_id: user_id });
+
+    request
+      .then((response) => {
+        if (response.status === 200 || response.status === 201) {
+          dispatch(setIsLoading(false));
+        }
+      })
+      .catch(() => dispatch(setIsError(true)));
+  };
+
+  useEffect(() => {
+    if (user_id) {
+      dispatch(setIsLoading(true));
+      _getUser(parseInt(user_id));
+      _getFollowersOfUser(parseInt(user_id));
+    }
+  }, [user_id]);
+
+  return (
+    <>
+      <Nav className='bg-primary' />
+      <Container className='flex flex-col m-10'>
+        <>
+          {state.category.isLoading && !state.category.isLoading && <Loader />}
+          {state.category.isError && (
+            <Notification
+              isSuccess={false}
+              title='An error has occurred. Please try again later.'
+              errorAction='refresh'
+            />
+          )}
+          {!state.category.isLoading && !state.category.isLoading && (
```

Comment:
Some conditions are hard to follow, can you put comments on explaining why this is being rendered like this?

Reply:
okay miss, will add comments